src/session/session_api.c

```diff
@@ -52,29 +52,58 @@ __wt_session_reset_cursors(WT_SESSION_IMPL *session, bool free_buffers)
  *     Sweep the cursor cache.
  */
 int
-__wt_session_cursor_cache_sweep(WT_SESSION_IMPL *session)
+__wt_session_cursor_cache_sweep(WT_SESSION_IMPL *session, bool big_sweep)
 {
+    WT_CONNECTION_IMPL *conn;
     WT_CURSOR *cursor, *cursor_tmp;
     WT_CURSOR_LIST *cached_list;
     WT_DECL_RET;
 #ifdef HAVE_DIAGNOSTIC
     WT_DATA_HANDLE *saved_dhandle;
 #endif
-    uint64_t now;
-    uint32_t position;
-    int i, t_ret, nbuckets, nexamined, nclosed;
+    uint64_t now, sweep_max, sweep_min;
+    uint32_t i, nclosed, position;
+    int t_ret, nbuckets, nexamined;
     bool productive;
 
     if (!F_ISSET(session, WT_SESSION_CACHE_CURSORS))
         return (0);
 
+    conn = S2C(session);
+
     /*
      * Periodically sweep for dead cursors; if we've swept recently, don't do it again.
+     *
+     * Each call of this sweep function visits all the cursors in some number of buckets used by the
+     * cursor cache. This determines if any of the visited cursors reference dead or dying data
+     * handles and should be fully closed and evicted from the cache. Evicting a cursor from the
+     * cursor cache has the important effect of freeing a reference to the associated data handle.
+     * Data handles can be closed and marked dead, but cannot be freed until all referencing
+     * sessions give up their references. So sweeping the cursor cache (for all sessions!) is a
+     * prerequisite for the connection data handle sweep to find handles that can be freed.
+     *
+     * So how many buckets to visit? This function is called in two ways. When a session reset is
```

Comment:
Instead of a rhetorical question, I suggest something like:
We determine the number of buckets to visit based on how the function is called....

Reply:
Fixed